src/main/java/com/github/kaktushose/jda/commands/dispatching/reply/ConfigurableReply.java

```diff
@@ -0,0 +1,198 @@
+    public ConfigurableReply(@NotNull MessageReply reply, @NotNull InteractionRegistry registry, @NotNull String runtimeId) {
+        super(reply);
+        this.interactionRegistry = registry;
+        this.runtimeId = runtimeId;
+    }
+
+    /// Constructs a new ConfigurableReply.
+    ///
+    /// @param reply the [ConfigurableReply] to copy
+    public ConfigurableReply(@NotNull ConfigurableReply reply) {
+        super(reply);
+        this.interactionRegistry = reply.interactionRegistry;
+        this.runtimeId = reply.runtimeId;
+    }
+
+    /// Whether to send ephemeral replies. Default value is `false`.
+    ///
+    /// Ephemeral messages have some limitations and will be removed once the user restarts their client.
+    /// Limitations:
+    /// - Cannot contain any files/ attachments
+    /// - Cannot be reacted to
+    /// - Cannot be retrieved
+    ///
+    /// **This will override both [GlobalReplyConfig] and any [ReplyConfig] annotation!**
+    ///
+    /// @param ephemeral `true` if to send ephemeral replies
+    /// @return the current instance for fluent interface
+    @NotNull
+    public ConfigurableReply ephemeral(boolean ephemeral) {
+        this.ephemeral = ephemeral;
+        return this;
+    }
+
+    /// Whether to keep the original components when editing a message. Default value is `true`.
+    ///
+    /// More formally, if editing a message and `keepComponents` is `true`, the original message will first be queried and
+    /// its components get added to the reply before it is sent.
+    ///
+    /// **This will override both [GlobalReplyConfig] and any [ReplyConfig] annotation!**
+    ///
+    /// @param editReply `true` if to keep the original components
+    /// @return the current instance for fluent interface
+    @NotNull
+    public ConfigurableReply editReply(boolean editReply) {
+        this.editReply = editReply;
+        return this;
+    }
+
+    /// Whether to edit the original message or to send a new one. Default value is `true`.
+    ///
+    /// The original message is always the very first reply that was sent. E.g. for a slash command event, which was
+    /// replied to with a text message and a button, the original message is that very reply.
+    ///
+    /// Subsequent replies to the same slash command event or the button event cannot be edited.
+    ///
+    /// **This will override both [GlobalReplyConfig] and any [ReplyConfig] annotation!**
+    ///
+    /// @param keepComponents `true` if to edit the original method
+    /// @return the current instance for fluent interface
+    @NotNull
+    public ConfigurableReply keepComponents(boolean keepComponents) {
+        this.keepComponents = keepComponents;
+        return this;
+    }
+
+    /// Adds an [ActionRow] to the reply and adds the passed components to it.
+    ///
+    /// The components will always be enabled and scoped. Use [#components(Components...)] if you want to modify these
```

Comment:
It's not clear what "scoped" means here.

Reply:
Oversight, pls rename it to "runtime-bound" like i already did in other places